### DISCUSSION TITLE
Explore the feasibility of using ColPali or ColQwen to enhance RAG functionalities

### DISCUSSION THREAD

**Discussion by NicolasDrapier** [open-webui/open-webui#7574] [Ideas]
I'd like to open a discussion about the possibility of using ColPali or ColQwen to enhance the functionalities of the RAG system. These tools have shown potential in similar applications and could provide significant benefits to the current setup.

**Comment by OM-EL** [comment]
it feels good to have a question and then find that someone asked the same question before.

**Comment by sancelot** [comment]
Hi, 
I made a pipeline that implements it.
https://github.com/sancelot/open-webui-multimodal-pipeline

**Reply by OM-EL** [reply]
I was trying to build this , feels greate to see other people have has same idea !

**Comment by OM-EL** [comment]
i have a question , 

do you thing integrating the indexing phase i the open webui is a good idea ( sustainable to hight loads of documents for a company grade deployment ? )

**Reply by sancelot** [reply]
At the moment my implementation reindexed the whole document because it was not possible to do it in openwebui for the qdrant db . I understand it needs improvement. but if we can overload db func in openwebui it would be nice
